### PR TITLE
[40] - View active auctions (buyer)

### PR DESCRIPTION
**Pull request by khasanrashidov**
### Where

[NL-40 View active auctions (buyer)](https://jjbcapital.atlassian.net/browse/NL-40)

### What

 Implementation of section that displays all the lots where buyer is participating at the moment. (or participated: won or lost)

### How

Developed both BE and FE to get buyer auctions (currently participating or participated) with pagination. There were slight changes in design, I decided to include and display some description of lot, because the card looked almost empty.

image

image

image

image

image

### Affected Area

Auctionify.API.ClientApp
Auctionify.Application.Features.Users.Queries (Make corrections in watchlist lots get method too, just noticed in the end some little bug)
Auctionify.UnitTests (for tests)

### PR Checklist

_Build_

- [x] Build Succeeded

_Unit Tests_

[x] All 79 Tests Passed, 0 Ignored, 0 Failed

### CODE REVIEW

**Review by otash8110 — APPROVED** [review]

**Review by SaidganiUsm — DISMISSED** [review]

File: src/Auctionify.API/ClientApp/src/app/components/buyer/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -1,61 +1,120 @@
...
+                if (!this.generalAuctionResponse.hasNext) {
+                    this.noMoreAuctionsToLoad = true;
+                }
+            });
+    }
+
     loadLotsInWatchlist() {
         this.apiClient.getLotsInWatchlist(0, 5).subscribe((lots) => {
             this.lots = lots;
+            console.log(this.lots);
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/buyer/dashboard/dashboard.component.ts:102]: Remove logs please

File: src/Auctionify.API/ClientApp/src/app/components/buyer/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -96,9 +156,32 @@ export class DashboardComponent implements OnInit {
...
                     this.loadLotsInWatchlist();
                 },
             });
         }
     }
+
+    determinePriceColor(auction: any) {
+        if (auction.lotStatus.name === 'Active') {
+            if (this.currentBuyerId === auction.bids[0].buyerId) {
+                this.priceColor = '#2b5293';
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/buyer/dashboard/dashboard.component.ts:175]: Move colors to constants with corresponded names

File: src/Auctionify.Application/Features/Users/Queries/GetBuyerAuctions/GetBuyerAuctionsQuery.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+				size: int.MaxValue,
+				cancellationToken: cancellationToken
+			);
+
+			var sortedBids = bids.Items.OrderByDescending(x => x.Id).ToList();
+
+			var lotIds = sortedBids.Select(x => x.LotId).ToList();
+
+			var lots = await _lotRepository.GetListAsync(
+				predicate: x => lotIds.Contains(x.Id),
```
Comment (TsvirkoNikita) [inline src/Auctionify.Application/Features/Users/Queries/GetBuyerAuctions/GetBuyerAuctionsQuery.cs:73]: Actually, you need to get only active auctions for that section. But as you already implemented some additional logic for showing auctions in other statuses, I could suggest adding sorting by status and show firstly active ones on this section.

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]

File: src/Auctionify.API/ClientApp/src/app/components/buyer/dashboard/dashboard.component.ts (hunk 1 above)
Comment (khasanrashidov) [reply src/Auctionify.API/ClientApp/src/app/components/buyer/dashboard/dashboard.component.ts:102]: Removed

File: src/Auctionify.Application/Features/Users/Queries/GetBuyerAuctions/GetBuyerAuctionsQuery.cs (hunk 3 above)
Comment (khasanrashidov) [reply src/Auctionify.Application/Features/Users/Queries/GetBuyerAuctions/GetBuyerAuctionsQuery.cs:73]: Yeah, I agree (about only active auctions), that is more logical. But in design there was one auction in which the buyer already won, and I clarified that from you (that: Is active auction the one in which the buyer is participating and participated?) That's why I decided to include some other auctions statuses.

Got it!) I will implement sorting.

Comment (khasanrashidov) [reply src/Auctionify.Application/Features/Users/Queries/GetBuyerAuctions/GetBuyerAuctionsQuery.cs:73]: I implemented sorting. First I am sorting (ordered by) active auctions, then sold auctions

**Review by TsvirkoNikita — APPROVED** [review]

**Review by otash8110 — APPROVED** [review]

**Review by SaidganiUsm — APPROVED** [review]

**Pull request merged**